test/AutoRest.TestServer.Tests/Common/Input/Source/SourceInputHelperTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.CSharp.Input.Source;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis;
+using NUnit.Framework;
+using System.Collections.Generic;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Common.Input;
+using AutoRest.CSharp.Output.Models.Types;
+using AutoRest.CSharp.Input;
+
+namespace AutoRest.CSharp.Utilities.Tests
+{
+    public class SourceInputHelperTests
+    {
+        private Compilation _compilation;
+
+        private INamedTypeSymbol _nullableIntSymbol;
+        private INamedTypeSymbol _listSymbol;
+        private INamedTypeSymbol _modelSymbol;
+
+        [OneTimeSetUp]
+        public void InitAsync()
+        {
+            var tree = CSharpSyntaxTree.ParseText(@"
+using System.Collections.Generic;
+
+namespace SourceInputHelperTests
+{
+    public class SourceInputMetadata
+    {
+        public class MetadataModel
+        {
+        }
+        public void Method(int? nullableInt, List<int> genericList, MetadataModel metadataModel)
+        {
+        }
+    }
+}");
+            var corlibLocation = typeof(object).Assembly.Location;
+            var references = new List<MetadataReference>
+            {
+                MetadataReference.CreateFromFile(corlibLocation)
+            };
+            _compilation = CSharpCompilation.Create("TestCode", syntaxTrees: new[] { tree }, references: references);
+
+            var members = _compilation.GetTypeByMetadataName("SourceInputHelperTests.SourceInputMetadata").GetMembers();
+            foreach (var member in members)
+            {
+                if (member is IMethodSymbol method && method.Name == "Method")
+                {
+                    foreach (var parameter in method.Parameters)
+                    {
+                        switch (parameter.Name)
+                        {
+                            case "nullableInt":
+                                _nullableIntSymbol = (INamedTypeSymbol)parameter.Type;
+                                break;
+                            case "genericList":
+                                _listSymbol = (INamedTypeSymbol)parameter.Type;
+                                break;
+                            case "metadataModel":
+                                _modelSymbol = (INamedTypeSymbol)parameter.Type;
+                                break;
+                        }
+                    }
+                }
+            }
+
+            Configuration.Initialize("Generated", "", "", new string[] { }, false, false, true, false, false, false, false, false, false, false, false, new List<string> { }, Configuration.UnreferencedTypesHandlingOption.RemoveOrInternalize, Configuration.AmbiguityHandlingOption.UseOverload, ".", null, new string[] { }, new List<string>(), new List<string>(), new List<string>(), null, null);
```

Comment:
Please use the positional arguments here so that it could be easier to update next time we introduce a new parameter in this method.